Automattic/wp-calypso
# Site Switcher: Use fuzzy search

#### Proposed Changes

Closes https://github.com/Automattic/wp-calypso/issues/67316. As an effort to standardize the search mechanisms between the sites page and instances of the site selector, let's use the fuzzy search functionality. It's yielding good results and producing a better UX.

It's blocked by https://github.com/Automattic/wp-calypso/pull/67881, a PR that adds the `title` computed field to the site excerpt query, which is used as one of the keys for fuzzy searching: `name` is not a very good index since it might be empty. `title` is a computed value between the `name` and the `slug`. Let's include all of them when fuzzy searching, along with the site URL.

#### Testing Instructions

1. Open wordpress.com;
2. Misstype some site name or URL in the site switcher;
3. Check that the "No sites found" message is displayed;
4. Apply this branch or open the dev site exposed in this PR;
5. Type the same search param in the site switcher;
6. Check that the site now appears, despite the typo;
7. Try searching in other instances of the site selector, such as the `/marketing/tools` page.

| `trunk` | This PR |
| ------- | -------- |
| ![image](https://user-images.githubusercontent.com/26530524/190545783-1d188290-f3d5-4c4b-a690-9441e0655a3c.png) | ![image](https://user-images.githubusercontent.com/26530524/190545806-4449b884-f987-486f-8678-af159e26afd3.png) |
